### PR TITLE
MBS-8436 add detekt plugin; apply and fix UnusedPrivateMember rule

### PR DESCRIPTION
**Pull request by dsvoronin**
image

### CODE REVIEW

File: subprojects/build.gradle.kts [hunk 1]
```diff
@@ -359,3 +378,21 @@ fun Project.configureJunit5Tests() {
...
+    include("**/*.kts")
+    exclude("**/resources/**")
+    exclude("**/build/**")
+    reports {
+        xml.enabled = false
+        html.enabled = false
+    }
+}
+
+tasks.named("check").dependsOn(detektAll)
```
Comment (dsvoronin) [inline subprojects/build.gradle.kts:411]: wiring to use standard `build` and `check` tasks chains

File: Makefile [hunk 2]
```diff
@@ -83,6 +83,12 @@ unit_tests:
 compile_tests:
 	./gradlew -p subprojects compileTestKotlin $(log_level)
 
+check:
+	./gradlew -p subprojects check
+
+detekt:
```
Comment (dsvoronin) [inline Makefile:89]: convenient `make detekt` for local iterations

File: subprojects/detekt.yml [hunk 3]
```diff
@@ -0,0 +1,689 @@
...
+
+output-reports:
+    active: true
+    exclude:
+    # - 'TxtOutputReport'
+    # - 'XmlOutputReport'
+    # - 'HtmlOutputReport'
+
+comments:
+    active: false
```
Comment (dsvoronin) [inline subprojects/detekt.yml:43]: All is copied from detekt default config and disabled (except UnusedPrivateMember);

Proposed pattern is to enable and fix valuable(for us) rules one by one and not use baseline mechanism

File: subprojects/build.gradle.kts [hunk 4]
```diff
@@ -359,3 +378,21 @@ fun Project.configureJunit5Tests() {
...
+    buildUponDefaultConfig = false
+    setSource(files(projectDir))
+    config.setFrom(files(project.rootDir.resolve("detekt.yml")))
+    include("**/*.kt")
+    include("**/*.kts")
+    exclude("**/resources/**")
+    exclude("**/build/**")
+    reports {
+        xml.enabled = false
+        html.enabled = false
```
Comment (eugene-krivobokov) [inline subprojects/build.gradle.kts:407]: Would it be easier to read html report in a build verdict?

File: subprojects/build.gradle.kts [hunk 5]
```diff
@@ -18,10 +20,12 @@ buildscript {
 }
 
 plugins {
+    base
```
Comment (sergioboi) [inline subprojects/build.gradle.kts:27]: What is the base?

File: subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt [hunk 6]
```diff
@@ -34,6 +34,7 @@ fun isPlaintext(buffer: Buffer): Boolean {
         val prefix = Buffer()
         val byteCount = if (buffer.size < 64) buffer.size else 64
         buffer.copyTo(prefix, 0, byteCount)
+        @Suppress("UnusedPrivateMember")
```
Comment (sergioboi) [inline subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt:37]: Didn't get the problem with that code

File: subprojects/detekt.yml (hunk 3 above)
Comment (eugene-krivobokov) [reply subprojects/detekt.yml:43]: How about adding this knowledge to this file?

Also, it'd be helpful to add a link to the reference: 
https://github.com/detekt/detekt/blob/master/detekt-core/src/main/resources/default-detekt-config.yml

File: subprojects/detekt.yml [hunk 7]
```diff
@@ -0,0 +1,689 @@
...
+    excludeCorrectable: false
+    weights:
+    # complexity: 2
+    # LongParameterList: 1
+    # style: 1
+    # comments: 1
+
+config:
+    validation: true
+    warningsAsErrors: false
```
Comment (sergioboi) [inline subprojects/detekt.yml]: True?

File: subprojects/detekt.yml [hunk 8]
```diff
@@ -0,0 +1,689 @@
...
+    ThrowsCount:
+        active: false
+        max: 2
+    TrailingWhitespace:
+        active: false
+    UnderscoresInNumericLiterals:
+        active: false
+        acceptableDecimalLength: 5
+    UnnecessaryAbstractClass:
+        active: false
```
Comment (sergioboi) [inline subprojects/detekt.yml:639]: could we use active = false by default?
I want to see what is enabled ASAP

File: subprojects/detekt.yml [hunk 9]
```diff
@@ -0,0 +1,689 @@
...
+        parameterPattern: '[a-z][A-Za-z0-9]*'
+        privateParameterPattern: '[a-z][A-Za-z0-9]*'
+        excludeClassPattern: '$^'
+        ignoreOverridden: true
+    EnumNaming:
+        active: true
+        excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/jsTest/**', '**/iosTest/**']
+        enumEntryPattern: '[A-Z][_a-zA-Z0-9]*'
+    ForbiddenClassName:
+        active: false
```
Comment (sergioboi) [inline subprojects/detekt.yml:375]: As I understand If on the top level of `naming` we have active=false then we don't need to duplicate active=false

File: subprojects/build.gradle.kts (hunk 5 above)
Comment (eugene-krivobokov) [reply subprojects/build.gradle.kts:27]: https://docs.gradle.org/current/userguide/base_plugin.html

File: subprojects/build.gradle.kts (hunk 4 above)
Comment (dsvoronin) [reply subprojects/build.gradle.kts:407]: don't know how to add it there; just using logs for now

File: subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt (hunk 6 above)
Comment (dsvoronin) [reply subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt:37]: `i` is unused

File: subprojects/detekt.yml (hunk 7 above)
Comment (dsvoronin) [reply subprojects/detekt.yml]: done (not affecting anything right now)

File: subprojects/detekt.yml (hunk 8 above)
Comment (dsvoronin) [reply subprojects/detekt.yml:639]: as far as i know detekt doesn't allow this

File: subprojects/detekt.yml (hunk 9 above)
Comment (dsvoronin) [reply subprojects/detekt.yml:375]: i only changed top level one's;
this is detekt defaults

File: subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt (hunk 6 above)
Comment (sergioboi) [reply subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt:37]: Could we change `i` to `_` and remove `Suppress`?

**Review by sergioboi — APPROVED** [review]

File: subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt (hunk 6 above)
Comment (dsvoronin) [reply subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt:37]: nope,

Comment (dsvoronin) [reply subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt:37]: nope, 
image

Comment (eugene-krivobokov) [reply subprojects/common/okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt:37]: It shows that we don't need an index here at all:

```kotlin
repeat(15) { ... }
```

**Pull request merged**